src/lib/integration_test_lib/test_error.ml

```diff
@@ -115,7 +115,7 @@ module Error_accumulator = struct
     let contexts_by_time =
       contextualized_errors |> String.Map.to_alist
       |> List.map ~f:(fun (ctx, errors) -> (errors.introduction_time, ctx))
-      |> Time.Map.of_alist_exn
+      |> Time.Map.of_alist_reduce ~f:(Printf.sprintf "%s, %s")
```

Comment:
Could you change this so to be a `string list Time.Map.t` instead? We ultimately want to iterate over each error individually, and the time indexing here is only used as a way to sort the errors by time. This change should be pretty simple to make and would improve how errors are printed when there is a time conflict.